instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py

```diff
@@ -339,6 +342,28 @@ def build_middleware_stack(self: Starlette) -> ASGIApp:
                 app,
             )
 
+            class ExceptionHandlerMiddleware:
+                def __init__(self, app):
+                    self.app = app
+
+                async def __call__(
+                    self, scope: Scope, receive: Receive, send: Send
+                ) -> None:
+                    try:
+                        await self.app(scope, receive, send)
+                    except Exception as exc:  # pylint: disable=broad-exception-caught
+                        span = get_current_span()
+                        span.record_exception(exc)
+                        span.set_status(
+                            Status(
+                                status_code=StatusCode.ERROR,
+                                description=f"{type(exc).__name__}: {exc}",
+                            )
+                        )
+                        raise
+
+            app.add_middleware(ExceptionHandlerMiddleware)
```

Comment:
I don't actually know where this middleware is going to go, it'd be clearer if this was inserted in  build_middleware_stack.